File: modules/41. RPC Protocols and data formats/Content/grpc-client-java/src/main/java/com/epam/grpc/GrpcClient.java
package com.epam.grpc;

import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import com.epam.grpc.PingPongServiceGrpc;
import com.epam.grpc.PingPong;

public class GrpcClient {
    public static void main(String[] args) {
        ManagedChannel channel = ManagedChannelBuilder
            .forTarget("localhost:8080")
            .usePlaintext()
            .build();
        PingPongServiceGrpc.PingPongServiceBlockingStub stub = PingPongServiceGrpc.newBlockingStub(channel);
        PingPong.PingRequest request = PingPong.PingRequest.newBuilder()
            .setMessage("Hello from Java GRPC Client")
            .setTimestamp(System.currentTimeMillis())
            .build();
        PingPong.PongResponse response = stub.pingPong(request);
        System.out.println("Got a response: '" + response.getMessage() + "' at " +response.getTimestamp());
        channel.shutdownNow();
    }
}

File: modules/41. RPC Protocols and data formats/Content/grpc-client-python/grpc-client.py
from os import times
from google.protobuf import message
import grpc
import ping_pong_pb2
import ping_pong_pb2_grpc
import time


def send_request():
	with grpc.insecure_channel('localhost:8080') as channel:
		stub = ping_pong_pb2_grpc.PingPongServiceStub(channel)
		request = ping_pong_pb2.PingRequest(
			message = 'Hello from Python GRPC Client',
			timestamp = int(time.time())
		)
		response = stub.pingPong(request)
		print(f'Got a response: "{response.message}" at {response.timestamp}')


if __name__ == '__main__':
	send_request()
